theia-python-docker/Dockerfile

```diff
@@ -1,22 +1,123 @@
+    gpg --batch --keyserver pgp.mit.edu --recv-keys "$key" || \
+    gpg --batch --keyserver keyserver.pgp.com --recv-keys "$key" || \
+    gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "$key" ; \
+    done
+
+ARG NODE_VERSION=10.15.3
+ENV NODE_VERSION $NODE_VERSION
+
+RUN ARCH= && dpkgArch="$(dpkg --print-architecture)" \
+    && case "${dpkgArch##*-}" in \
+    amd64) ARCH='x64';; \
+    ppc64el) ARCH='ppc64le';; \
+    s390x) ARCH='s390x';; \
+    arm64) ARCH='arm64';; \
+    armhf) ARCH='armv7l';; \
+    i386) ARCH='x86';; \
+    *) echo "unsupported architecture"; exit 1 ;; \
+    esac \
+    && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-$ARCH.tar.xz" \
+    && curl -SLO --compressed "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
+    && gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc \
+    && grep " node-v$NODE_VERSION-linux-$ARCH.tar.xz\$" SHASUMS256.txt | sha256sum -c - \
+    && tar -xJf "node-v$NODE_VERSION-linux-$ARCH.tar.xz" -C /usr/local --strip-components=1 --no-same-owner \
+    && rm "node-v$NODE_VERSION-linux-$ARCH.tar.xz" SHASUMS256.txt.asc SHASUMS256.txt \
+    && ln -s /usr/local/bin/node /usr/local/bin/nodejs
+
+ENV YARN_VERSION 1.13.0
+
+RUN set -ex \
+    && for key in \
+    6A010C5166006599AA17F08146C2130DFD2497F5 \
+    ; do \
+    gpg --batch --keyserver ipv4.pool.sks-keyservers.net --recv-keys "$key" || \
+    gpg --batch --keyserver pool.sks-keyservers.net --recv-keys "$key" || \
+    gpg --batch --keyserver pgp.mit.edu --recv-keys "$key" || \
+    gpg --batch --keyserver keyserver.pgp.com --recv-keys "$key" || \
+    gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "$key" ; \
+    done \
+    && curl -fSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz" \
+    && curl -fSLO --compressed "https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc" \
+    && gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz \
+    && mkdir -p /opt/yarn \
+    && tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/yarn --strip-components=1 \
+    && ln -s /opt/yarn/bin/yarn /usr/local/bin/yarn \
+    && ln -s /opt/yarn/bin/yarn /usr/local/bin/yarnpkg \
+    && rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz
+
+# Developer tools
+
+## Git and sudo (sudo needed for user override)
+RUN apt-get -y install git sudo
+
+# Python Tools
 RUN apt-get update \
-    && apt-get install -y python python-dev python-pip \
+    && apt-get install -y python-dev python-pip \
+    && apt-get install -y python3-dev python3-pip \
     && apt-get clean && rm -rf /var/cache/apt/* && rm -rf /var/lib/apt/lists/* && rm -rf /tmp/*
 
-RUN pip install \
-    python-language-server \
-    flake8 \
-    autopep8
+## User account
+RUN adduser --disabled-password --gecos '' theia && \
+    adduser theia sudo && \
```

Comment:
The previous version of this image I think did not have sudo capability? If so adding it could be considered a breaking change by some users, that may be running it in environment where this is not permitted. 

Any particular reason to include this now?

Reply:
I believe the only reason it is now included is since it was copied over from 
the `theia-full-docker` image. Do we expect to not be breaking in the images? 
(I thought they were for education purposes like examples on how to build their own images or for us to test functionality present in the main repo)

If it causes issues I can try to remove it and see if it still works correctly.

Reply:
I see the following comment:

> Git and sudo (sudo needed for user override)

https://github.com/theia-ide/theia-apps/pull/297/files#diff-db44fcad56aa1590715c9f3963aa5a86R77

Reply:
I do not have a strong opinion either way.

@jgroom33 As the original contributor of this image, do you have an opinion on whether it should have sudo available?

Reply:
its nice to have sudo access to `apt install ...` but the goal of the image should probably be to have a more strict image creation. sudo should not be allowed
It should be the responsibility of the person doing the container packaging to modify this standard docker image and add the necessary packages for their use case.
Modifying an existing image that defaults to no sudo is easier than solving the problem of how to remove sudo and follow the standard.

Reply:
@jgroom33 good points, thanks for the feedback. 

@vince-fugnitto let's remove sudo

Reply:
> @vince-fugnitto let's remove sudo

@marcdumais-work we should remove `sudo`, and do the following updates 
as well? https://github.com/theia-ide/theia-apps/pull/297#discussion_r377925621

Reply:
> @marcdumais-work we should remove `sudo`, and do the following updates
> as well? [#297 (comment)](https://github.com/theia-ide/theia-apps/pull/297#discussion_r377925621)

:+1: